File: backend_dotnet/fruit_api/fruit_api/Models/OrderVoucher.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("OrderVouchers")]  // Sửa từ Order_vouchers thành OrderVouchers
public class OrderVoucher
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("orderVoucherId")]  // Sửa từ order_voucher_id thành orderVoucherId
    public string OrderVoucherId { get; set; } = null!;

    [Required]
    [Column("orderId")]  // Sửa từ order_id thành orderId
    public string OrderId { get; set; } = null!;

    [Required]
    [Column("voucherId")]  // Sửa từ voucher_id thành voucherId
    public string VoucherId { get; set; } = null!;

    [Required]
    [Column("discountAmount", TypeName = "decimal(12,2)")]  // Sửa từ discount_amount thành discountAmount
    public decimal DiscountAmount { get; set; }

    // Navigation properties
    public Order? Order { get; set; }
    public Voucher? Voucher { get; set; }
}

File: frontend_flutter/fruit_market/lib/models/OrderVoucher.dart
import 'Voucher.dart';

class OrderVoucher {
  final String orderVoucherId;
  final String orderId;
  final String voucherId;
  final double discountAmount;

  /// nếu API trả luôn voucher
  final Voucher? voucher;

  OrderVoucher({
    required this.orderVoucherId,
    required this.orderId,
    required this.voucherId,
    required this.discountAmount,
    this.voucher,
  });

  /// JSON -> Object
  factory OrderVoucher.fromJson(Map<String, dynamic> json) {
    return OrderVoucher(
      orderVoucherId: json['orderVoucherId'],
      orderId: json['orderId'],
      voucherId: json['voucherId'],
      discountAmount: (json['discountAmount'] as num).toDouble(),
      voucher:
          json['voucher'] != null ? Voucher.fromJson(json['voucher']) : null,
    );
  }

  /// Object -> JSON
  Map<String, dynamic> toJson() {
    return {
      'orderVoucherId': orderVoucherId,
      'orderId': orderId,
      'voucherId': voucherId,
      'discountAmount': discountAmount,
      'voucher': voucher?.toJson(),
    };
  }
}